Tools/automation.ps1

```diff
@@ -0,0 +1,790 @@
+                $AzStorageAccountKey  = $(Get-AzStorageAccountKey -ResourceGroupName $AzResourceGroup -Name $AzStorageAccountName)[0].Value
+
+                ## Retrieves the required information from the previously created Azure objects. Values will be used to generate required information for the Azure Keyvault.
+                $CosmosAccountEndpointValue       = ConvertTo-SecureString -String $($(Get-AzCosmosDBAccount -ResourceGroupName $AzResourceGroup).DocumentEndpoint) -AsPlainText -Force
+                $CosmosAccountKeyValue            = ConvertTo-SecureString -String $($(Get-AzCosmosDBAccountKey -ResourceGroupName $AzResourceGroup -Name $CosmosAccountName).PrimaryMasterKey) -AsPlainText -Force
+                $AzStorageAccountConnectionString = ConvertTo-SecureString -String "DefaultEndpointsProtocol=https;AccountName=$AzStorageAccountName;AccountKey=$AzStorageAccountKey;EndpointSuffix=$AzEndpointSuffix" -AsPlainText -Force
+
+                ## Adds the Azure Storage Account Connection String to the Keyvault
+                Write-Host "      Creating Keyvault Secret for Azure Storage Account Connection String."
+                $Result = Set-AzKeyVaultSecret -VaultName $AzKeyVaultName -Name $AzStorageAccountKeyName -SecretValue $AzStorageAccountConnectionString
+    
+                ## Adds the Azure Cosmos Account Endpoint URL to the Keyvault
+                Write-Host "      Creating Keyvault Secret for Azure CosmosDB Connection String."
+                $Result = Set-AzKeyVaultSecret -VaultName $AzKeyVaultName -Name $CosmosAccountEndpointKeyName -SecretValue $CosmosAccountEndpointValue
+    
+                ## Adds the Azure Cosmos Primary Account Key to the Keyvault
+                Write-Host "      Creating Keyvault Secret for Azure CosmosDB Primary Key."
+                $Result = Set-AzKeyVaultSecret -VaultName $AzKeyVaultName -Name $CosmosAccountKeyKeyName -SecretValue $CosmosAccountKeyValue
+    
+                ## Create base object of the Azure Function, generating reference object ID for Keyvault
+                Write-Host "    Creating base Azure Function object."
+                $Result = New-AzResourceGroupDeployment -ResourceGroupName $AzResourceGroup -TemplateFile $Object.TemplatePath -TemplateParameterFile $Object.ParameterPath -Mode Incremental -ErrorAction SilentlyContinue
+            }
+    
+            ## Creates the Azure Resource
+            Write-Host "    Creating $($Object.ObjectType) following the ARM Parameter File..."
+            $Result = New-AzResourceGroupDeployment -ResourceGroupName $AzResourceGroup -TemplateFile $Object.TemplatePath -TemplateParameterFile $Object.ParameterPath -Mode Incremental -ErrorAction SilentlyContinue -ErrorVariable objerror
+    
+            ## Sets a sleep of 10 seconds after object creation to allow Azure to update creation status, and mark as "running"
+            Start-Sleep -Seconds 10
```

Comment:
variable based on load. Might be nice if we checked it the resource was in the expected state before proceeding.

Reply:
I'll take a look and see what is available to pull a status from.

Reply:
+1

Reply:
Hey @arjun-msft and @jamespik,

Let me ask this question. What would be the best approach to this. Scenario being:

Object is created, status is marked as "Complete" but a reference to the "completed" object fails from another object if referenced within the first few seconds of it's creation.